client/client.go

```diff
@@ -47,7 +47,8 @@ type Proxy struct {
 // with Toxiproxy. Endpoint is the address to the proxy (e.g. localhost:8474 if
 // not overriden)
 func NewClient(endpoint string) *Client {
-	if !strings.HasPrefix(endpoint, "http://") {
+	protocolProvided, _ := regexp.MatchString(`^https?://`, endpoint)
```

Comment:
Generally it's better to create a regex via `regexp.MustCompile()` and then calling the `MatchString` method. This ensures we don't ignore a possible error in the regex.